playbooks/robusta_playbooks/popeye.py

```diff
@@ -0,0 +1,215 @@
+class PopeyeReport(BaseModel):
+    score: int    
+    grade: str
+    sanitizers:  Optional[List[PopeyeSection]]
+    errors:  Optional[List[str]]
+
+
+class GroupedIssues(BaseModel):
+    issues = []
+    level: int = 0
+
+def levelToString(level: int) -> str:
+    if level == 1:
+        return "I"
+    elif level == 2:
+        return "W"
+    elif level == 3:
+        return "E"
+    else:
+        return "OK"
+
+def scanRowContentToString(row: ScanReportRow) -> str:
+    txt = f"**{row.container}**\n" if row.container else "" 
+    for i in row.content:
+        txt+= f"{levelToString(i['level'])} {i['message']}\n"
+    
+    return txt
+
+class PopeyeParams(ProcessParams):
+    """
+    :var image: The popeye container image to use for the scan.
+    :var timeout: Time span for yielding the scan.
+    :var args: Popeye cli arguments.
+    :var spinach: Spinach.yaml config file to supply to the scan.
+    :var serviceAccountName: The account name to use for the Popeye scan job.
+    """
+
+    image: str = "derailed/popeye" 
+    serviceAccountName: str = f"{RELEASE_NAME}-runner-service-account"
+    timeout = 300
+    args: str = "-s no,ns,po,svc,sa,cm,dp,sts,ds,pv,pvc,hpa,pdb,cr,crb,ro,rb,ing,np,psp"
+    spinach: str = """\
+popeye:
+    excludes: 
+        apps/v1/daemonsets:
+        - name: rx:kube-system
+        apps/v1/deployments:
+        - name: rx:kube-system
+        v1/configmaps:
+        - name: rx:kube-system
+        v1/pods:
+        - name: rx:kube-system
+        v1/services:
+        - name: rx:kube-system
+        v1/namespaces:
+        - name: kube-system"""
+
+
+
+
+def group_issues_list(issues: List[Issue]) -> Dict[str,GroupedIssues]:
+    groupedIssues: Dict[str, GroupedIssues] = defaultdict(lambda: GroupedIssues())
+    for issue in issues:
+        group = groupedIssues[issue.group]
+        group.issues.append({"level":issue.level, "message":issue.message})
+        group.level = max(group.level, issue.level)
+
+    return groupedIssues
+
+
+@action
+def popeye_scan(event: ExecutionBaseEvent, params: PopeyeParams):
+    """
+    Displays a popeye scan report.
+    """
+
+    spec = PodSpec(
+        serviceAccountName=params.serviceAccountName,
+        containers=[
+            Container(
+                name=to_kubernetes_name(params.image),
+                image=params.image,
+                command=["/bin/sh", "-c", f"echo '{params.spinach}' > ~/spinach.yaml && popeye -f ~/spinach.yaml {params.args} -o json --force-exit-zero"],
+            )
+        ],
+        restartPolicy="Never",
+    )
+
+    start_time = end_time = datetime.now()
+    popeye_scan = scan = {} 
+    logs = None
+    
+    try:
+        logs = RobustaJob.run_simple_job_spec(spec,"popeye_job",params.timeout)
+        scan = json.loads(logs)
+        end_time = datetime.now() 
+        popeye_scan = PopeyeReport(**scan['popeye'])
+    except JSONDecodeError as e:
```

Comment:
I'm not sure we should create a finding each time the scan fails.
Might be ok in the early stages of the feature